src/backend/distributed/commands/table.c

```diff
@@ -652,11 +652,17 @@ PostprocessAlterTableSchemaStmt(Node *node, const char *queryString)
 	ObjectAddress tableAddress = GetObjectAddressFromParseTree((Node *) stmt, true);
 
 	/* check whether we are dealing with a sequence here */
-	if (get_rel_relkind(tableAddress.objectId) == RELKIND_SEQUENCE)
+	char relKind = get_rel_relkind(tableAddress.objectId);
+	if (relKind == RELKIND_SEQUENCE)
 	{
 		stmt->objectType = OBJECT_SEQUENCE;
 		return PostprocessAlterSequenceSchemaStmt((Node *) stmt, queryString);
 	}
+	else if (relKind == RELKIND_VIEW)
```

Comment:
Q1: why is this happening before `!ShouldPropagate` ?  

Q2: Also, we have `PostprocessAlterViewSchemaStmt` as well, why do we have this code? Is it because we can do `ALTER TABLE view SET SCHEMA`? Even that seems to go through `PostprocessAlterViewSchemaStmt`

I'm in favor of not allowing such weird syntax even if PG allows. As one of the most senior PG commiters noted for PG [here](https://www.postgresql.org/message-id/flat/F681D64D-8C81-4665-8ABE-D8DCF1A8E511%40amazon.com#57e747c1f3b7904d44fb0a5ee7a296e8):

> In short: no, I do not agree that this is a bug to be fixed.  Perhaps
we should have done things differently years ago, but it's too late to
redefine it.

I think it is time for us to build things in a way we prefer. We shouldn't really need to support this weird approach. Isn't it confusing?

Reply:
> Q1: why is this happening before !ShouldPropagate ?

because `PostprocessAlterViewSchemaStmt` does that anyway.